diff --git a/packages/env/src/file.ts b/packages/env/src/file.ts
index cd84dbff..b02d7961 100644
--- a/packages/env/src/file.ts
+++ b/packages/env/src/file.ts
@@ -51,7 +51,7 @@ export const getFileConfig = () => {
       S3_BUCKET: z.string().optional(),
       S3_ENABLE_PATH_STYLE: z.boolean(),
 
-      S3_ENDPOINT: z.string().url().optional(),
+      S3_ENDPOINT: z.string().url().optional().or(z.literal('')),
       S3_PREVIEW_URL_EXPIRE_IN: z.number(),
       S3_PUBLIC_DOMAIN: z.string().optional(),
       S3_REGION: z.string().optional(),
